apps/core/faviconmanager.cpp

```diff
@@ -193,12 +195,14 @@ void FaviconManager::grabIcon(const QString &type, DeclarativeWebPage *webPage,
         QObject::disconnect(*dataConn);
         if (dataFetcher->hasAcceptedTouchIcon()) {
             qCDebug(lcFavoritesLog) << "Storing favicon for" << type;
+qWarning() << "XXXXXXXXXXXXXXX got favicon result: " << dataFetcher->data() << " for url: " << webPage->url().toString() << " with type: " << type;
```

Comment:
interestingly, I never see this occurring.  In my testing (e.g. loading google.com / jolla.com / twitter.com / amazon.com they never return a favicon result here... which seems strange?

Reply:
I get the same results as you. I did some really brief tests, moving your `grabIcon()` call from `WebView.onUrlChanged` and into `WebView.onLoadedChanged`. I had more luck in this case (see this debug output: [gecko.log](https://github.com/sailfishos/sailfish-browser/files/6629460/gecko.log)).

I'm not at all sure that `onLoadedChanged` is the correct place to put it, but I guess the important point is that the favicon url is read from the html file, so it can only be safely used after at least the main html file has been loaded by the browser.

Reply:
If there's a way to solve this inside FaviconManager, rather than relying on `grabIcon` being called in the correct place, that would be really nice.

Reply:
Thanks, I tried doing it with:

```
    if (!webPage->domContentLoaded()) {
        QObject *contextObj = new QObject(this);
        connect(webPage, &DeclarativeWebPage::domContentLoadedChanged,
                contextObj, [this, type, webPage, size, contextObj] {
                    this->grabIcon(type, webPage, size);
                    contextObj->deleteLater();
                }, Qt::UniqueConnection);
        return;
    }
```

and then calling grabIcon() in the onUrlChanged handler, but it didn't work (it appears to trigger too "soon" and we get some strange results like the favicon for the PREVIOUS site being stored for the next browse location).

For now, I think triggering grabIcon() from within `onLoadedChanged: if (loaded) ... ` makes the most sense.

Reply:
Thanks for trying. I agree, if there's no easy way to make it work universally, then better to implement something bespoke that will work fine for the history.